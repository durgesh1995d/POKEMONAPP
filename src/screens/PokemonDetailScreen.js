import React from 'react';
import { SafeAreaView } from 'react-native';
import PokemonDetail from '../component/PokemonDetail';

const PokemonDetailScreen = ({ route }) => {
  return (
    <SafeAreaView>
      <PokemonDetail route={route} testID="PokemonDetail" />
    </SafeAreaView>
  );
};

export default PokemonDetailScreen;
