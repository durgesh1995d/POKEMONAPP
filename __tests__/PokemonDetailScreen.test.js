import React from 'react';
import { render } from '@testing-library/react-native';
import PokemonDetailScreen from '../src/screens/PokemonDetailScreen';
import PokemonDetail from '../src/component/PokemonDetail';

// Mock the PokemonDetail component
jest.mock('../src/component/PokemonDetail', () => 'PokemonDetail');

describe('PokemonDetailScreen', () => {
  it('renders PokemonDetailScreen correctly', () => {
    const route = { params: { name: 'pikachu' } }; // Mock route prop

    const { getByTestId, toJSON } = render(
      <PokemonDetailScreen route={route} />
    );

    expect(getByTestId('PokemonDetail')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
