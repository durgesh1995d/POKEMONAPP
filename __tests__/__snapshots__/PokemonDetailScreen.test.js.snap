// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetailScreen renders PokemonDetailScreen correctly 1`] = `
<RCTSafeAreaView>
  <PokemonDetail
    route={
      {
        "params": {
          "name": "pikachu",
        },
      }
    }
    testID="PokemonDetail"
  />
</RCTSafeAreaView>
`;
